Юркин Павел БПМ-18-3


Пирсон: Задача 6

В некоторой местности в течение 300 суток регистрировалась среднесуточная температура воздуха. В итоге наблюдений было получено эмпирическое распределение, приведенное в таблице. В первом столбце указан интервал температуры в градусах, во втором столбце – частота ni , т.е. количество дней, среднесуточная температура которых принадлежит этому интервалу).

![alt text](prsn6.png "Title")

Требуется при уровне значимости α=0,05 проверить гипотезу о том, что  среднесуточная температура воздуха распределена равномерно.  

$H_0$ - Распределение среднесуточной темпиратуры воздуха равномерное.<br>
$H_1$ - Распределение среднесуточной темпиратуры воздуха НЕ равномерное.

In [1]:
#Середина интервала
temp = [-35, -25, -15, -5, 5, 15, 25, 35]
#Частоты
freq = [25, 40, 30, 45, 40, 46, 48, 26] 

In [2]:
def mean(x, freq):
    return sum([(x[i]*freq[i]) for i in range(len(x))])/sum(freq)      

In [3]:
def var(x, freq, mean):
    return sum([freq[i]*(x[i] - mean)**2 for i in range(len(x))])/sum(freq)

In [4]:
smpmean = mean(temp, freq)
smpvar = var(temp, freq, mean(temp, freq))

## Предполагаемая плотность распеделения

In [5]:
a = smpmean - (3 * smpvar)**(0.5) 
b = smpmean + (3 * smpvar)**(0.5)
pdf = 1/(b - a)

## Теоретические частоты

In [6]:
temp_f = [-40, -30, -20, -10, 0, 10, 20, 30, 40]
n_1 = sum(freq) * pdf * (temp_f[1] - a)
theory_freq = [n_1]

for i in range(len(freq) - 2):
    theory_freq.append(sum(freq) * pdf * (temp_f[i+1] -  temp_f[i]))
theory_freq.append(sum(freq) * pdf * (b - temp_f[-2]))

In [7]:
theory_freq

[21.8872228811985,
 40.67072289485762,
 40.67072289485762,
 40.67072289485762,
 40.67072289485762,
 40.67072289485762,
 40.67072289485762,
 34.088439749655784]

In [8]:
def pearson(f, tf):
    return sum([(f[i]-tf[i])**2/tf[i] for i in range(len(f))]) 

In [9]:
#наблюдаемое значение
pearson(freq, theory_freq)

7.663660161900434

k = s - 2 - 1  = 5, при уровне значимости 0.05, по таблице критических точек
критическое значение = 11,1
так как наблюдаемое значение меньше критического (7.66 < 11.1), $H_0$ не отклоняется

Розенбаум: Задача 1
![alt text](rosenbaum.jpg "Title")

$H_0$ - уровень вербального интеллекта по методике Векслера в группе физиков не превосходит группу психологов

$H_1$ - уровень вербального интеллекта по методике Векслера в группе психолгово превосходит группу физиков

In [10]:
#физики
phy = [132, 134, 124, 132, 135, 132, 131, 132, 121, 127, 136, 129, 136, 136]
#психологи
psy = [126, 127, 132, 120, 119, 126, 120, 123, 120, 116, 123, 115]

n1 = 0
for i in range(len(phy)):
    if phy[i] > max(psy):
        n1+=1
n2 = 0
for i in range(len(psy)):
    if psy[i] < min(phy):
        n2 +=1
#наблюдаемое значение равно 11        
print(n1 + n2)


11


Критическое значение при уровне значимости 0.05 равно 7

Так как критическое значение меньше налюдаемого (7 < 11) гипотеза $H_0$ отвергается 

Критерий знаков: 4
![alt text](znaki.jpg "Title")

$H_0$ - Эффективность не увеличилась после использования метода Y

$H_1$ - Эффективность увеличилась после использования метода Y

In [11]:
x = [18, 10, 7, 15, 14, 11, 13]
y = [15, 20, 10, 8, 16, 10, 19] # 7, 15, 14, 29
diff = [y[i] - x[i] for i in range(len(x))]
diff

[-3, 10, 3, -7, 2, -1, 6]

In [66]:
morethan0 = [a if a > 0 else 0 for a in diff]
morethan0 = len(morethan0) - morethan0.count(0)
lessthan0 = [a if a < 0 else 0 for a in diff]
lessthan0 = len(lessthan0) - lessthan0.count(0)

4 3


In [67]:
#наблюдаемое значение
observed = lessthan0 if morethan0 > lessthan0 else morethan0 
observed

3

Критическое значение при уровне значимости 0.05 и выборки длинны 7 равно 0.
Так как наблюдаемое значение меньше критического (3 < 7), гипотеза $H_0$ не отклоняется

Уилксон: 2
![alt text](yuilk.jpg "Title")

$H_0$ - Введение новой экономической политики НЕ привело к увеличению производительности

$H_1$ - Введение новой экономической политики привело к увеличению производительности

In [40]:
x = [52, 51, 48, 52, 54, 50, 51, 51, 52, 52, 53, 56, 51, 50]
y = [44, 47, 57, 54, 39, 65, 46, 51, 58, 46, 62, 52, 65, 47]

In [57]:
yx = [[abs(y[i] - x[i]), y[i] - x[i]] for i in range(len(x))]
yx.sort() 

In [58]:
dups = set()
duplicates = set()
for i in range(len(yx)):
    if yx[i][0] not in dups:
        dups.add(yx[i][0])
    else:
        duplicates.add(yx[i][0])

cnt = 2
yx[0].append(1)
for i in range(1, len(yx)):
    if yx[i][0] in duplicates and yx[i-1][0] != yx[i][0]:
        yx[i].append(cnt + 0.5)
    elif yx[i][0] in duplicates and yx[i-1][0] == yx[i][0]:
        yx[i].append(yx[i-1][2])
    elif yx[i-1][0] in duplicates and yx[i][0] not in duplicates:
        yx[i].append(cnt)
    else:
        yx[i].append(cnt)
    cnt += 1

In [64]:
negative_tendention_sum = sum([yx[i][2] if yx[i][1] < 0 else 0 for i in range(len(yx))])
#наблюдаемое значение
negative_tendention_sum

48.0

Критическое значение при уровне значимости 0.05 и размером выборки 14 равно 25, так как наблюдаемое значение больше, чем критическое (48 > 25), гипотеза $H_0$ не отклоняется.